Std/Data/BitVec/Bitblast.lean

```diff
@@ -157,3 +158,58 @@ theorem adc_correct (x y : BitVec w) (c : Bool) :
 
 theorem add_as_adc (w : Nat) (x y : BitVec w) : x + y = (adc x y false).snd := by
   simp [adc_correct]
+
+/-! ### Negation -/
+
+/-- Bitwise 1's complement implemented via `iunfoldr`. -/
+def bit_not (x : BitVec w) : BitVec w :=
+  ((iunfoldr fun (i : Fin w) c => (c, !(x.getLsb i))) ()).snd
+
+/-- Bitwise 2's complement implemented via `iunfoldr`. -/
+def bit_neg (x : BitVec w) : BitVec w := (adc (bit_not x) (BitVec.ofNat w 1) false).snd
```

Comment:
What is the motivation for these definitions? Is the implementation via `iunfoldr` more efficient? I don't understand why there is a need for two implementations of the same function.

Reply:
The idea is to implement a bit-blasted version of each `BitVec` operation and show it is equivalent to the original definition. The advantage of this is that SMT solvers use bitblasting to convert such operations into a boolean formula which can be sent to a SAT solver. Our aim is to develop a tactic that integrates SMT solvers with Lean in this project: [lean-smt](https://github.com/ufmg-smite/lean-smt).

Reply:
I understand. It's unusual to have defs that aren't meant to be run. Would it be sufficient to have a theorem that says `-x =  (adc (bit_not x) (BitVec.ofNat w 1) false).snd`? It's fine if not, so long as it's clear that these are alternate implementations not meant to be used in code.